test/extensions/filters/network/mongo_proxy/proxy_test.cc

```diff
@@ -42,7 +43,9 @@ class MockDecoder : public Decoder {
   MOCK_METHOD(void, onData, (Buffer::Instance & data));
 };
 
-class TestStatStore : public Stats::IsolatedStoreImpl {
+// Note: there is a MockStore in test/common/stats/mocks.h but it appears to
+// only allow for one mock counter, so we must create our own.
```

Comment:
Can we fix the public mock?

Reply:
Can I put a TODO here to explore unifying that?

Reply:
Yes that's fine.

Reply:
resolved now.